test/e2e/specs/editor/various/unsynced-pattern.spec.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Unsynced pattern', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'create a new unsynced pattern via the block options menu', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'A useful paragraph to reuse' },
+		} );
+
+		// Create an unsynced pattern from the paragraph block.
+		await editor.showBlockToolbar();
+		await page
+			.getByRole( 'toolbar', { name: 'Block tools' } )
+			.getByRole( 'button', { name: 'Options' } )
+			.click();
+		await page.getByRole( 'menuitem', { name: 'Create pattern' } ).click();
+
+		const createPatternDialog = page.getByRole( 'dialog', {
+			name: 'Create pattern',
+		} );
+		await createPatternDialog
+			.getByRole( 'textbox', { name: 'Name' } )
+			.fill( 'My unsynced pattern' );
+		await page.getByLabel( 'Synced' ).click();
+
+		await page.keyboard.press( 'Enter' );
+		const content = await editor.getEditedPostContent();
+
+		// Check that the block content is still the same. If the pattern was added as synced
+		// the content would be wrapped by a pattern block.
+		expect( content ).toBe(
+			`<!-- wp:paragraph -->
+<p>A useful paragraph to reuse</p>
+<!-- /wp:paragraph -->`
+		);
```

Comment:
It's recommended to avoid snapshot testing as suggested in https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/overusing-snapshots.md. I think this is a good example that we can use explicit assertion to avoid this:

```js
const before = await editor.getBlocks(); // Or `editor.getEditedPostContent()`

// Create the pattern...

await expect.poll(
  editor.getBlocks,
  'The block content should be the same after converting to an unsynced pattern'
).toEqual( before );
```

Reply:
also good to know - the dangers of copying and pasting from another spec instead of reading the docs. Updated.